contracts/Tokens/Prime/Prime.sol

```diff
@@ -0,0 +1,908 @@
+     * @param user account for which we need to accrue rewards
+     * @param vToken the market for which we need to accrue rewards
+     */
+    function _executeBoost(address user, address vToken) internal {
+        if (!markets[vToken].exists || !tokens[user].exists) {
+            return;
+        }
+
+        accrueInterest(vToken);
+        interests[vToken][user].accrued += _interestAccrued(vToken, user);
+        interests[vToken][user].rewardIndex = markets[vToken].rewardIndex;
+    }
+
+    /**
+     * @notice Update total score of user and market. Must be called after changing account's borrow or supply balance.
+     * @param user account for which we need to update score
+     * @param market the market for which we need to score
+     */
+    function _updateScore(address user, address market) internal {
+        if (!markets[market].exists || !tokens[user].exists) {
+            return;
+        }
+
+        uint score = _calculateScore(market, user);
+        markets[market].sumOfMembersScore = markets[market].sumOfMembersScore - interests[market][user].score + score;
+        interests[market][user].score = score;
+    }
+
+    /**
+     * @notice Distributes income from market since last distribution
+     * @param vToken the market for which to distribute the income
+     */
+    function accrueInterest(address vToken) public {
+        if (!markets[vToken].exists) revert MarketNotSupported();
+
+        address underlying = _getUnderlying(vToken);
+
+        IPrimeLiquidityProvider _primeLiquidityProvider = IPrimeLiquidityProvider(primeLiquidityProvider);
+
+        uint totalIncomeUnreleased = IProtocolShareReserve(protocolShareReserve).getUnreleasedFunds(
+            comptroller,
+            IProtocolShareReserve.Schema.SPREAD_PRIME_CORE,
+            address(this),
+            underlying
+        );
+
+        uint256 distributionIncome = totalIncomeUnreleased - unreleasedPSRIncome[underlying];
+
+        _primeLiquidityProvider.accrueTokens(underlying);
+        uint256 totalAccruedInPLP = _primeLiquidityProvider.tokenAmountAccrued(underlying);
+        uint256 unreleasedPLPAccruedInterest = totalAccruedInPLP - unreleasedPLPIncome[underlying];
+
+        distributionIncome += unreleasedPLPAccruedInterest;
+
+        if (distributionIncome == 0) {
+            return;
+        }
+
+        unreleasedPSRIncome[underlying] = totalIncomeUnreleased;
+        unreleasedPLPIncome[underlying] = totalAccruedInPLP;
+
+        uint256 delta;
+        if (markets[vToken].sumOfMembersScore > 0) {
+            delta = ((distributionIncome * EXP_SCALE) / markets[vToken].sumOfMembersScore);
+        }
+
+        markets[vToken].rewardIndex = markets[vToken].rewardIndex + delta;
+    }
+
+    /**
+     * @notice Returns boosted interest accrued for a user
+     * @param vToken the market for which to fetch the accrued interest
+     * @param user the account for which to get the accrued interest
```

Comment:
```suggestion
     * @param user the account for which to get the accrued interest
     * @return interestAccrued the number of underlying tokens accrued by the user since the last accrual
```